thirdparty/faiss/faiss/IndexIVFFlat.cpp

```diff
@@ -514,6 +596,192 @@ void IndexIVFFlat::reconstruct_from_offset(
     memcpy(recons, invlists->get_single_code(list_no, offset), code_size);
 }
 
+std::unique_ptr<IVFFlatIteratorWorkspace> IndexIVFFlat::getIteratorWorkspace(
+        const float* query_data,
+        const IVFSearchParameters* ivfsearchParams) const {
+    return std::make_unique<IVFFlatIteratorWorkspace>(
+            query_data, ivfsearchParams);
+}
+
+std::optional<std::pair<float, idx_t>> IndexIVFFlat::getIteratorNext(
```

Comment:
my editor is complaining `namespace "std" has no member "optional"`. let's `#include <optional>`.

it's a good practice to Include What You Use. We don't currently have a proper tool set up to automate/enforce this but I try to do my best manually. 

Reference: https://github.com/include-what-you-use/include-what-you-use

Reply:
added.